### PR TITLE
Rewrote SPARQL page specifications

### PR DESCRIPTION
**Pull request by mihailradkov**
* The specification now makes use of the repo creation and deletion via REST commands
* Simplified selectors and functions
* Added proper checks

Added Cypress command for URL importing via REST call.
Moved the RDF snippet import command to import-commands.js

Added a few selectors in the sparql templates for easier selection in
the Cypress tests.

### CODE REVIEW

**Comment by mihailradkov** [comment]
https://travis-ci.com/Ontotext-AD/graphdb-workbench/builds/123728466#L1514

Failures in the reworked spec

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 1]
```diff
@@ -1,807 +1,1034 @@
...
+                disableSameAs: {
+                    value: false
                 }
-            });
-        deleteCurrentTab();
-    });
+            }
+        }));
+
+        it('Test execute query including inferred with ruleset "OWL-Horst (Optimized)"', () => {
```
Comment (mihailradkov) [inline test-cypress/integration/sparql/sparql.menu.spec.js]: These 2 can be combined into single spec.

**Comment by svilenvelikov** [comment]
We should decide what are we going to do with the sonar issues.

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 2]
```diff
@@ -1,807 +1,1034 @@
...
-        cy.selectRepoS(testRepo).wait(1000);
+            // Press the pen icon to edit the custom query created earlier
+            clickEditSaveQueryFromPopup(savedQueryName);
 
-        cy.navigateToPageS('SPARQL');
+            getPopover().should('not.exist');
+            getSavedQueryForm().should('be.visible');
 
-        let queryToExecute = 'INSERT DATA { <urn:a> <http://a/b> <urn:b> . <urn:b> <http://a/b> <urn:c> . }';
+            // TODO: Can make use of .within?
```
Comment (svilenvelikov) [inline test-cypress/integration/sparql/sparql.menu.spec.js]: yes

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 3]
```diff
@@ -1,807 +1,1034 @@
...
-        cy.get('@uncheckIncludeInferred').click();
+            // Close edit saved query dialog
+            getSubmitSavedQueryBtn().click();
 
-        // Click on the Run button
-        cy.get('div#runButton > #wb-sparql-runQuery').as('rerunQuery');
+            // Click on the saved queries icon
+            getSavedQueriesPopupBtn().click();
+            // Press the trash bin to delete the custom query created earlier
+            clickDeleteSaveQueryFromPopup(savedQueryName);
```
Comment (svilenvelikov) [inline test-cypress/integration/sparql/sparql.menu.spec.js]: `deleteSaveQueryFromPopup` ? We have to discuss about the style and follow it in order to have consistency.

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 4]
```diff
@@ -1,807 +1,1034 @@
...
 
-        // Go to SPARQL and execute a SELECT query without a limit
-        cy.executeCustomQuery(defaultQueryWithoutLimit);
+        it('Test URL to current query', () => {
+            const query = 'SELECT ?s ?p ?o WHERE {?s ?p ?o .} LIMIT 100';
+            typeQuery(query);
 
-        // Verify that there are 7,065 results
-        cy.verifyQueryResultsS('1,000 of 7,065');
+            // Press the link icon to generate a link for a query
```
Comment (svilenvelikov) [inline test-cypress/integration/sparql/sparql.menu.spec.js]: If the function is like `generateLinkForQuery` the comment will be redundant

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 5]
```diff
@@ -1,807 +1,1034 @@
...
+            getQueryLinkBtn().click();
 
-        // Disable the inference from the ">>" icon on the right of the SPARQL editor.
-        cy.get('#inference').as('disableInference');
+            // TODO: Test with tab name -> should open such tab if it was deleted
```
Comment (svilenvelikov) [inline test-cypress/integration/sparql/sparql.menu.spec.js]: Maybe we should directly create tasks for these?!

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 6]
```diff
@@ -1,807 +1,1034 @@
...
+        // Using force because the textarea is not visible
+        getQueryTextArea().type('{ctrl}a{backspace}', {force: true});
+    }
+
+    function typeQuery(query) {
+        clearQuery();
+        getQueryTextArea().type(query, {force: true, parseSpecialCharSequences: false});
+    }
+
+    function pasteQuery(query) {
```
Comment (svilenvelikov) [inline test-cypress/integration/sparql/sparql.menu.spec.js]: Well it appeared we have a paste functionality already. Why we don't use it more extensively?

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 7]
```diff
@@ -1,807 +1,1034 @@
...
+            // Check that they are horizontal
+            cy.get('#sparql-content > div').should('have.class', 'row');
+            cy.get('#yasr').should('not.have.class', 'vertical');
+
+            // Change the orientation to vertical
+            getViewOrientationButton().click();
+
+            // Verify that all elements are visible and properly displayed.
+            cy.get('#queryEditor').should('be.visible');
+            cy.get('#yasr-inner').should('be.visible');
```
[the remark this answers is not in the document]
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: Yes

File: test-cypress/integration/sparql/sparql.menu.spec.js (hunk 5 above)
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: Or at least add them in the acceptance sheet.

File: test-cypress/integration/sparql/sparql.menu.spec.js (hunk 6 above)
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: I tended using it for bigger queries.

Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: If we add exclusive test that verifies correct typing, we could replace most of it with the paste method (excluding the automatic prefix insertion which doesn't work with pasting :) )

File: test-cypress/integration/sparql/sparql.menu.spec.js (hunk 4 above)
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: Almost all of the present comments were left from before the refactoring.
For this one I agree it's unnecessary.

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 8]
```diff
@@ -1,807 +1,1034 @@
...
+            verifyTabName(2, secondTabName);
+        });
 
-        // Click on the Run button
-        cy.get('div#runButton > #wb-sparql-runQuery').as('runQuery');
+        it('Test close a tab', () => {
+            getNewTabButton()
+                .click()
+                .click()
+                .click();
```
[the remark this answers is not in the document]
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: I had such methods originally, but I felt they were not following Cypress ideas. So I removed most of them.
Honestly, in context of Cypress, I cannot find balance between extracting logic and repetition (with the idea of simplicity and readability) ...

**Comment by mihailradkov** [comment]
https://github.com/Ontotext-AD/graphdb-workbench/pull/35#issuecomment-522832314

Well, I suggest to leave them until we start working on reducing the technical debt. The current issues are not caused by my changes but are result of the status quo.

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 9]
```diff
@@ -1,807 +1,1034 @@
...
+            getSameAsButton()
+                .find('.icon-sameas-on')
+                .should('be.visible');
+
+            typeQuery(selectQuery);
+            executeQuery();
+            verifyResultsPageLength(2);
+
+            getSameAsButton()
+                .click()
```
[the remark this answers is not in the document]
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: Okay but what about the icon visibly check ? Two calls, e.g. `enableSameAs()` + `getSameAsButton().find(...).should(...)` ? OR return the button chain from `enableSameAs()` ? The second becomes a little ambiguous, the thing you didn't like about the import specification.

[the remark this answers is not in the document]
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: Some possible simplifications

- `getSavedQueryField().clear().type(QUERY_FOR_SAVING_MODIFIED)` -> `typeSavedQuery()`
-  `getTabs().should('have.length', 1)` -> `verifyTabsLength()` ?
- `getResultPages().should('have.length', 2)` -> `verifyResultPages()`

[the remark this answers is not in the document]
Comment (svilenvelikov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: Sounds good

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 10]
```diff
@@ -1,807 +1,1113 @@
...
         '} limit 1001';
 
-    let defaultQuery = 'select * where { \n' +
-        '\t?s ?p ?o .\n' +
-        '} limit 100 ';
+    function createRepoAndVisit(repoOptions = {}) {
+        repositoryId = 'sparql-' + Date.now();
+        repoOptions.id = repositoryId;
+        cy.createRepository(repoOptions);
+        cy.warmRepositoryNamespaces(repositoryId);
```
Comment (svilenvelikov) [inline test-cypress/integration/sparql/sparql.menu.spec.js]: Should we always wait for the namespaces after creating the repo?

File: test-cypress/integration/sparql/sparql.menu.spec.js [hunk 11]
```diff
@@ -1,807 +1,1113 @@
...
+    }
+
+    function clearQuery() {
+        // Using force because the textarea is not visible
+        getQueryTextArea().type('{ctrl}a{backspace}', {force: true});
+    }
+
+    function typeQuery(query) {
+        clearQuery();
+        getQueryTextArea().type(query, {force: true, parseSpecialCharSequences: false});
```
Comment (svilenvelikov) [inline test-cypress/integration/sparql/sparql.menu.spec.js]: I guess all `force:true ` are needed everywhere where the hidden textarea is operated?

**Comment by mihailradkov** [comment]
Squashed some of the commits

File: test-cypress/integration/sparql/sparql.menu.spec.js (hunk 11 above)
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: Yes

File: test-cypress/integration/sparql/sparql.menu.spec.js (hunk 10 above)
Comment (mihailradkov) [reply test-cypress/integration/sparql/sparql.menu.spec.js]: Not sure so that's why I didn't add it by default after each repo creation.

**Comment by svilenvelikov** [comment]
There is failing test "1) Test download query results in Supported formats"

**Pull request merged**